diff --git a/1-1.patch b/1-1.patch
index f020c8f6..e69de29b 100644
--- a/1-1.patch
+++ b/1-1.patch
@@ -1,28 +0,0 @@
-diff --git a/strings/split.py b/strings/split.py
-index ed194ec6..31d1c1d3 100644
---- a/strings/split.py
-+++ b/strings/split.py
-@@ -21,13 +21,17 @@ def split(string: str, separator: str = " ") -> list:
- 
-     split_words = []
- 
-+    sep_len = len(separator)
-     last_index = 0
--    for index, char in enumerate(string):
--        if char == separator:
--            split_words.append(string[last_index:index])
--            last_index = index + 1
--        if index + 1 == len(string):
--            split_words.append(string[last_index : index + 1])
-+    i = 0
-+    while i <= len(string) - sep_len:
-+        if string[i : i + sep_len] == separator:
-+            split_words.append(string[last_index:i])
-+            last_index = i + sep_len
-+            i = last_index
-+            continue
-+        i += 1
-+    split_words.append(string[last_index:])
-     return split_words
- 
- 
diff --git a/strings/split.py b/strings/split.py
index 31d1c1d3..6e5e8117 100644
--- a/strings/split.py
+++ b/strings/split.py
@@ -3,22 +3,33 @@ def split(string: str, separator: str = " ") -> list:
     Will split the string up into all the values separated by the separator
     (defaults to spaces)
 
-    >>> split("apple#banana#cherry#orange",separator='#')
+    >>> split("apple#banana#cherry#orange", separator="#")
     ['apple', 'banana', 'cherry', 'orange']
 
+    >>> split("apple<>banana<>cherry", separator="<>")
+    ['apple', 'banana', 'cherry']
+
     >>> split("Hello there")
     ['Hello', 'there']
 
-    >>> split("11/22/63",separator = '/')
+    >>> split("11/22/63", separator="/")
     ['11', '22', '63']
 
-    >>> split("12:43:39",separator = ":")
+    >>> split("12:43:39", separator=":")
     ['12', '43', '39']
 
-    >>> split(";abbb;;c;", separator=';')
+    >>> split(";abbb;;c;", separator=";")
     ['', 'abbb', '', 'c', '']
+
+    >>> split("abc", separator="")
+    Traceback (most recent call last):
+        ...
+    ValueError: empty separator
     """
 
+    if separator == "":
+        raise ValueError("empty separator")
+
     split_words = []
 
     sep_len = len(separator)
